ext/RMagick/rmimage.c

```diff
@@ -10875,22 +10877,23 @@ Image_recolor(VALUE self, VALUE color_matrix)
 
     order = (unsigned long)sqrt((double)(len + 1.0));
 
-    // RecolorImage sets the ExceptionInfo and returns a NULL image if an error occurs.
+    exception = AcquireExceptionInfo();
```

Comment:
I think this line does not need to move.
It is easier to understand when it is executed immediately before use.

Reply:
I moved it a bit earlier because this needs to be done in ImageMagick 7. The diff with ImageMagick 7 will be cleaner if this is done first.

Reply:
Please split cleanup and purpose for IM7 to ease review.

Reply:
No problem. Will remove it from this PR first and then create a new one when this has been merged.